src/json-api/json-api-hapi-plugin.ts

```diff
@@ -589,20 +640,69 @@ export class JsonApiHapiPlugin {
     return reply(this.getEntity('notification', api => api.getProjectNotificationConfigurations(projectId)));
   }
 
-  public async postNotificationConfigurationHandler(request: Hapi.Request, reply: Hapi.IReply) {
-    const config = request.payload.data.attributes;
-
-    config.teamId = config.teamId || null;
-    config.projectId = config.projectId || null;
+  private async authorizeNotificationConfiguration(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      reply(request.payload.data.attributes);
+    } catch (error) {
+      reply(Boom.badRequest());
+    }
+  }
 
-    if (!config.teamId && !config.projectId) {
-      throw Boom.badRequest('teamId or projectId should be defined');
+  private async authorizeTeamOrProjectAccess(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      const config = request.pre.filter;
```

Comment:
Good if that is possible! 

I would still suggest adding the more specific and type for the first parameter and doing early destructuring for `teamId` and `projectId`.